scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram_transducer.py

```diff
@@ -13,257 +13,420 @@
     wer_dist_best = cer_dist_best = 0
     words_count = 0
     chars_count = 0
+    sample_idx = 0
     if preds_output_file:
-        out_file = open(preds_output_file, 'w')
-
-    it = tqdm(range(len(all_hypotheses)), desc=f"Beam search decoding...", ncols=120,)
-
-    for sample_idx in it:
-        hypotheses = all_hypotheses[sample_idx]
-        target = target_transcripts[sample_idx]
-        target_split_w = target.split()
-        target_split_c = list(target)
-        words_count += len(target_split_w)
-        chars_count += len(target_split_c)
-        wer_dist_min = cer_dist_min = 10000
-        if not isinstance(hypotheses, list):
-            hypotheses = [hypotheses]
-        for candidate_idx, candidate in enumerate(hypotheses):
-            pred_text = candidate.text
-            pred_split_w = pred_text.split()
-            wer_dist = editdistance.eval(target_split_w, pred_split_w)
-            pred_split_c = list(pred_text)
-            cer_dist = editdistance.eval(target_split_c, pred_split_c)
-
-            wer_dist_min = min(wer_dist_min, wer_dist)
-            cer_dist_min = min(cer_dist_min, cer_dist)
-
-            if candidate_idx == 0:
-                # first candidate
-                wer_dist_first += wer_dist
-                cer_dist_first += cer_dist
-
-            if preds_output_file:
-                out_file.write('{}\t{}\n'.format(pred_text, candidate.score))
-        wer_dist_best += wer_dist_min
-        cer_dist_best += cer_dist_min
+        out_file = open(preds_output_file, 'w', encoding='utf_8', newline='\n')
+
+    if progress_bar:
+        if cfg.decoding_strategy == "greedy_batch":
+            description = "Greedy_batch decoding.."
+        else:
+            description = f"{cfg.decoding_strategy} decoding with bw={beam_width}, ba={beam_alpha}, ma={maes_prefix_alpha}, mg={maes_expansion_gamma}, hat_ilmw={hat_ilm_weight}"
+        it = tqdm(range(int(np.ceil(len(all_probs) / beam_batch_size))), desc=description, ncols=120)
+    else:
+        it = range(int(np.ceil(len(all_probs) / beam_batch_size)))
+    for batch_idx in it:
+        # disabling type checking
+        probs_batch = all_probs[batch_idx * beam_batch_size : (batch_idx + 1) * beam_batch_size]
+        probs_lens = torch.tensor([prob.shape[-1] for prob in probs_batch])
+        with torch.no_grad():
+            packed_batch = torch.zeros(len(probs_batch), probs_batch[0].shape[0], max(probs_lens), device='cpu')
+
+            for prob_index in range(len(probs_batch)):
+                packed_batch[prob_index, :, : probs_lens[prob_index]] = torch.tensor(
+                    probs_batch[prob_index].unsqueeze(0), device=packed_batch.device, dtype=packed_batch.dtype
+                )
+            best_hyp_batch, beams_batch = model.decoding.rnnt_decoder_predictions_tensor(
+                packed_batch, probs_lens, return_hypotheses=True,
+            )
+        if cfg.decoding_strategy == "greedy_batch":
```

Comment:
Why do we need / want to support greedy decoding in a beam search script? Drop support for it, we have transcribe_speech script for that.